src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,134 @@
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Arm extends SubsystemBase {
+
+    private final static Arm INSTANCE = new Arm();
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    private final CANSparkMax
+        masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+        followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+        masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+        followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public Command getSetTargetAngleCommand(Rotation2d angle){
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(angle),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {},
+                () -> false,
+                this
+        );
+    }
+
+    public Command getSetTargetElevatorCommand(Rotation2d angle){
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(angle),
+                this::setTargetElevatorFromProfile,
+                (interrupted) -> {},
+                () -> false,
+                this
+        );
+    }
+
+    private void setTargetAngleFromProfile(){
+        if (angleMotorProfile == null){
+            stopAngleMotors();
+            return;
+        }
+
+        final TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        masterAngleMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
+    }
+
+    private void setTargetElevatorFromProfile(){
+        if (elevatorMotorProfile == null){
+            stopElevatorMotors();
+            return;
+        }
+
+        final TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        masterElevatorMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle){
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(),0),
+                new TrapezoidProfile.State(getAngleMotorPositionDegrees(), getAngleMotorVelocity())
+        );
+
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(Rotation2d targetAngle){
```

Comment:
This shouldn't have an angle param.

Reply:
but then what do I use instead in the command?